lib/Minz/ThemeExtension.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+abstract class Minz_ThemeExtension extends Minz_Extension {
+	/**
+	 * Get the list of files to use while loading the theme.
+	 *
+	 * @return array
+	 */
+	abstract protected function getThemeFiles();
+
+	public function isDefault() {
+		return false;
+	}
+
+	public function init() {
+	}
+
+	public function install() {
+		$current_user = Minz_Session::param('currentUser');
+		$themes_path = THEME_PATH . "/{$current_user}";
+
+		// Create the directory for the user themes
+		if (!file_exists($themes_path) && false === mkdir($themes_path)) {
+			return "Cannot create {$themes_path} directory.";
+		}
+
+		// Create the theme directory
+		$theme_path = "{$themes_path}/{$this->getName()}";
+		if (!file_exists($theme_path) && false === mkdir($theme_path)) {
+			return "Cannot create {$theme_path} directory.";
+		}
+
+		// Copy the content of the extension theme to the user theme
+		$filepath = join_path($this->getPath(), 'static');
```

Comment:
Is a full copy of the theme necessary?

Reply:
Probably not. But for that, we need to include a way of knowing which files to include.
There's nothing like that at the moment.

Reply:
And if some kind of user data is necessary, it would be nice to put it under `./data/users/xxx/...`

Reply:
I am not there yet :)